be/src/connector/connector.cpp

```diff
@@ -20,30 +20,20 @@ void ConnectorManager::put(const std::string& name, std::unique_ptr<Connector> c
 }
 
 ConnectorManager* ConnectorManager::default_instance() {
-    return &global_default_instance;
+    return &_global_default_instance;
 }
 
-void ConnectorManager::init() {
-    ConnectorManager* cm = default_instance();
-    cm->put("hive", std::make_unique<HiveConnector>());
-}
+const std::string Connector::HIVE = "hive";
 
-Status DataSource::open(RuntimeState* state) {
-    // TODO: common operations
-    RETURN_IF_ERROR(do_open(state));
-    return Status::OK();
-}
+class ConnectorManagerInit {
+public:
+    ConnectorManagerInit() {
+        ConnectorManager* cm = ConnectorManager::default_instance();
```

Comment:
Is it safe to reference `_global_default_instance` here, because the order of initialization of `_global_default_instance` and `_init` is not guaranteed